### PR TITLE
Update prompt and model for watsonx summarization

### PR DESCRIPTION
**Pull request by jwm4**
Context: https://github.ibm.com/watson-engagement-advisor/wea-backlog/issues/60756

Major changes:
- Model used for the watsonx kit is now Granite v2 instruct.
- Prompt template used for the watsonx kit is updated based on input from Andrew Freed a few months ago with additional re-engineering for Granite v2 instruct.  (Sorry this has taken so long to get done!)
- README updates:
  -  More details about the prompt template being used + how and why users might want to do their own prompt engineering.
  - Technical details were moved to the watsonx.ai example section since that is the focus now.
  - README now uses the phrase "watsonx.ai" more often instead of just "watsonx" since there are an assortment of watsonx branded products now.
  - Sample screenshot updated to show actual Granite v2 instruct output.

### CODE REVIEW

File: integrations/extensions/starter-kits/language-model-summarization/summarization-watsonx-actions.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
-  "name": "summarization_watsonx-action",
+  "name": "Summarization murdockj 2024-1-action",
```
Comment (anirudhaBapat) [inline integrations/extensions/starter-kits/language-model-summarization/summarization-watsonx-actions.json:2]: Want to change this name back to summarization_watsonx-action?

**Review by anirudhaBapat — APPROVED** [review]
Tested on PSTG slot https://us-south.assistant.test.watson.cloud.ibm.com/crn%3Av1%3Astaging%3Apublic%3Aconversation%3Aus-south%3Aa%2F24d97bb9c71a4d019ca21d195231d92d%3Aa7935a19-c2c6-4d7f-8baa-7ea98897b078%3A%3A/assistants/1f82a83a-c195-4756-8ad7-ed89fb491754/actions/variables/custom 

It works as expected. I guess this slot has a service desk setup so not directly giving an error on escalate to agent. Here is the screenshot if you want to update your screenshot to this. 

Screenshot 2024-01-18 at 9 58 43 AM

File: integrations/extensions/starter-kits/language-model-summarization/summarization-watsonx-actions.json (hunk 1 above)
Comment (jwm4) [reply integrations/extensions/starter-kits/language-model-summarization/summarization-watsonx-actions.json:2]: Good idea -- will do!

**Comment by jwm4** [comment]
The image in the PR came from prod, so I think I will stick with it since I think it is closest to what new users will see.

**Review by anirudhaBapat — APPROVED** [review]
LGTM!

**Review by ethanwinters — APPROVED** [review]

**Pull request merged**